tests/Unit/Http/Validation/Rules/UsernameTest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Engelsystem\Test\Unit\Http\Validation\Rules;
+
+use Engelsystem\Application;
+use Engelsystem\Config\Config;
+use Engelsystem\Http\Validation\Rules\Username;
+use Engelsystem\Test\Unit\ServiceProviderTest;
+
+class UsernameTest extends ServiceProviderTest
+{
+    private Username $subject;
+
+    private Config $config;
+
+    public function setUp(): void
+    {
+        $this->subject = new Username();
+
+        $app = $this->getApp(['get']);
+        $this->config = new Config([]);
+        $app->method('get')->willReturnMap([['config', $this->config]]);
+        Application::setInstance($app);
+    }
+    /**
+     * @return array<string,array{string,bool}>
+     */
+    public function provideValidateWithDefaultConfigTestData(): array
+    {
+        return [
+            'empty string' => ['', false],
+            'max length exceeded' => [str_repeat('1', 25), false],
+            'invalid chars' => ['abc!!!', false],
+            'valid Greek letters' => ['λουκάνικο', true],
+            'min length valid' => ['a', true],
+            'valid with all chars' => ['abc123_-. jkl', true],
+            'valid with accents' => ['café', true],
+            'max length valid' => [str_repeat('a', 24), true],
+        ];
+    }
```

Comment:
is the same as `provideValidateWithCustomConfigTestData`. only `'valid Greek letters'` and `'min length valid'` are swapped

Reply:
Will be gone when moving the default to the config.